esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -3198,14 +3199,48 @@ void CWsWorkunitsEx::getFileResults(IEspContext &context, const char *logicalNam
     if (!df)
         throw makeStringExceptionV(ECLWATCH_FILE_NOT_EXIST, "Cannot find file %s.", logicalName);
 
+    __uint64 resultMaxSize = getWUResultMaxSize(context, logicalName);
     Owned<IResultSetFactory> resultSetFactory = getSecResultSetFactory(context.querySecManager(), context.queryUser(), context.queryUserId(), context.queryPassword());
     Owned<INewResultSet> result(resultSetFactory->createNewFileResultSet(df, cluster));
     if (!filterBy || !filterBy->length())
-        appendResultSet(context, buf, result, resname.str(), start, count, total, bin, xsd, context.getResponseFormat(), nullptr, wuResultMaxSize);
+        appendResultSet(context, buf, result, resname.str(), start, count, total, bin, xsd, context.getResponseFormat(), nullptr, resultMaxSize);
     else
     {
         Owned<INewResultSet> filteredResult = createFilteredResultSet(result, filterBy);
-        appendResultSet(context, buf, filteredResult, resname.str(), start, count, total, bin, xsd, context.getResponseFormat(), nullptr, wuResultMaxSize);
+        appendResultSet(context, buf, filteredResult, resname.str(), start, count, total, bin, xsd, context.getResponseFormat(), nullptr, resultMaxSize);
+    }
+}
+
+__int64 CWsWorkunitsEx::getWUResultMaxSize(IConstWorkUnit *cw)
+{
+    if (!cw) //ECL WU does not exist.
+        return wuResultMaxSize; //Use the default.
+
+    __int64 wuResultMaxSizeMB = cw->getDebugValueInt64("wuresultmaxsizemb", 0);
+    if (wuResultMaxSizeMB > 0)
+        return wuResultMaxSizeMB * 0x100000;
+    else
+        return wuResultMaxSize;
+}
+
+__int64 CWsWorkunitsEx::getWUResultMaxSize(IEspContext &context, const char *logicalName)
```

Comment:
given other comments, I am not sure either of these function should remain, at least in their current form.